app/pb/references.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package org.couchers.api.references;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+service References {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  rpc ListReferences(ListReferencesReq) returns (ListReferencesRes) {
+    // Returns a paginated list of references satisfying some filtering criterion
+  }
+
+  rpc WriteFriendReference(WriteFriendReferenceReq) returns (Reference) {
+    // Write a reference for your friend. Raises INVALID_ARGUMENT if you
+    // try to write for yourself, or NOT_FOUND if the user doesn't exist.
+    // Raises FAILED_PRECONDITION if you try to write a reference for someone you already wrote a friend reference for.
+  }
+
+  rpc WriteHostRequestReference(WriteHostRequestReferenceReq) returns (Reference) {
+    // Write a reference for someone after having hosted/surfed with them. Raises NOT_FOUND if the request doesn't exist.
+  }
+
+  rpc AvailableWriteReferences(AvailableWriteReferencesReq) returns (AvailableWriteReferencesRes) {
+    // Return the reference types that is, at the moment, possible to write about a user.
+  }
+
+  rpc ListPendingReferencesToWrite(google.protobuf.Empty) returns (ListPendingReferencesToWriteRes) {
+    // Returns a list of references that you should write about others, e.g. after surfing or hosting
+  }
+}
+
+enum ReferenceType {
+  REFERENCE_TYPE_FRIEND = 0;  // We met on an event but haven't surfed or hosted
+  REFERENCE_TYPE_SURFED = 1;  // The author have surfed, the referent has hosted
+  REFERENCE_TYPE_HOSTED = 2;  // The author have hosted, the referent has surfed
+}
+
+message Reference {
+  int64 reference_id = 1;
+
+  int64 from_user_id = 2;
+  int64 to_user_id = 3;
+
+  ReferenceType reference_type = 4;
+  string text = 5;  // plain text
+  google.protobuf.Timestamp written_time = 6;  // not exact
+
+  // if the reference is about the current user, this tells which host request it's about
+  int64 host_request_id = 7;
+}
+
+message ListReferencesReq {
+  // at least one of these two must be set, zero means "any"
+  uint64 from_user_id = 1;
+  uint64 to_user_id = 2;
+
+  // empty filter means include all, otherwise only those in the filter
+  repeated ReferenceType reference_type_filter = 3;
+
+  uint32 page_size = 4;
+  string page_token = 5;
+}
+
+message ListReferencesRes {
+  repeated Reference references = 1;
+
+  string next_page_token = 2;
+}
+
+message WriteFriendReferenceReq {
+  int64 to_user_id = 1;
+  string text = 2;  // plain text
+  bool was_safe = 3;
+  int32 rating = 4;
```

Comment:
Based on the mockups, I think this should have `bool was_comfortable` (maybe appropriate is a better word than comfortable, though?) and rating should be a float from 0-1. I am going to ask for clarification in the slack.

Reply:
@aapeliv

Reply:
Done.